src/app/clusters/webrtc-transport-provider-server/webrtc-transport-provider-server.cpp

```diff
@@ -0,0 +1,474 @@
+            ctx, [this](HandlerContext & subCtx, const auto & req) { HandleProvideICECandidates(subCtx, req); });
+        break;
+
+    case Commands::EndSession::Id:
+        CommandHandlerInterface::HandleCommand<Commands::EndSession::DecodableType>(
+            ctx, [this](HandlerContext & subCtx, const auto & req) { HandleEndSession(subCtx, req); });
+        break;
+
+    default:
+        // Mark unrecognized command as UnsupportedCommand
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::UnsupportedCommand);
+        break;
+    }
+}
+
+// Helper functions
+WebRTCSessionStruct * WebRTCTransportProviderServer::FindSession(uint16_t sessionId)
+{
+    for (auto & session : mCurrentSessions)
+    {
+        if (session.id == sessionId)
+        {
+            return &session;
+        }
+    }
+
+    return nullptr;
+}
+
+void WebRTCTransportProviderServer::AddOrUpdateSession(const WebRTCSessionStruct & session)
+{
+    // Search for a session with the same ID
+    for (auto & existing : mCurrentSessions)
+    {
+        if (existing.id == session.id)
+        {
+            // Update it in-place
+            existing = session;
+            return;
+        }
+    }
+
+    // If not found, add a new entry
+    mCurrentSessions.push_back(session);
+}
+
+void WebRTCTransportProviderServer::RemoveSession(uint16_t sessionId)
+{
+    // Erase-Remove idiom
+    mCurrentSessions.erase(std::remove_if(mCurrentSessions.begin(), mCurrentSessions.end(),
+                                          [sessionId](const WebRTCSessionStruct & s) { return s.id == sessionId; }),
+                           mCurrentSessions.end());
+}
+
+uint16_t WebRTCTransportProviderServer::GenerateSessionID()
+{
+    static uint16_t lastSessionId = 0;
+    uint16_t candidateId          = lastSessionId++;
+
+    // Handle wrap-around per spec
+    if (lastSessionId > 65534)
+        lastSessionId = 0;
+
+    // Ensure uniqueness
+    while (FindSession(candidateId) != nullptr)
+    {
+        candidateId = lastSessionId++;
+        if (lastSessionId > 65534)
+            lastSessionId = 0;
+    }
+
+    return candidateId;
+}
+
+// Command Handlers
+void WebRTCTransportProviderServer::HandleSolicitOffer(HandlerContext & ctx, const Commands::SolicitOffer::DecodableType & req)
+{
+    if (!req.videoStreamID.HasValue() && !req.audioStreamID.HasValue())
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::ConstraintError);
+        return;
+    }
+
+    // Prepare arguments for Delegate::HandleSolicitOffer using SolicitOfferRequestArgs struct
+    Delegate::SolicitOfferRequestArgs args;
+    args.id                 = GenerateSessionID();
+    args.streamUsage        = req.streamUsage;
+    args.videoStreamId      = req.videoStreamID;
+    args.audioStreamId      = req.audioStreamID;
+    args.iceServers         = req.ICEServers;
```

Comment:
Do we need to pass the TLV encoded structs to the Delegate?

Reply:
We are already passing the decoded attributes (videoStreamID, audioStreamID, ICEServers, etc.) to the delegate. The delegate can then work with these typed fields without worrying about TLV details.